lib/puma/cluster.rb

```diff
@@ -179,7 +187,8 @@ def check_workers(refork = false)
       timeout_workers
       wait_workers
       cull_workers
-      spawn_workers
+      promote_mold if @options[:mold_worker]
+      spawn_workers unless @pending_phased_restart
```

Comment:
This doesn't make any sense if we're about to start a phased restart; we'll create new workers only to immediately replace them with workers from the new phase.